### PR TITLE
bpo-35284: Fix the error handling in the compiler's compiler_call()

### PR DESCRIPTION
**Pull request by ZackerySpytz**
compiler_call() needs to check if an error occurred during the
maybe_optimize_method_call() call.

https://bugs.python.org/issue35284

### CODE REVIEW

File: Python/compile.c [hunk 1]
```diff
@@ -3622,8 +3622,10 @@ maybe_optimize_method_call(struct compiler *c, expr_ty e)
 static int
 compiler_call(struct compiler *c, expr_ty e)
 {
-    if (maybe_optimize_method_call(c, e) > 0)
-        return 1;
+    int ret = maybe_optimize_method_call(c, e);
+    if (ret >= 0) {
```
Comment (brettcannon) [inline Python/compile.c:3917]: Why the change from `> 0` to `>= 0`?

**Comment by serhiy-storchaka** [comment]
Good catch! :clap: It is hard to notice this error. :male_detective: How did you find it?

Do you mind to add a comment to `maybe_optimize_method_call()` about returning values? And please describe the error in more detail on the tracker.

File: Python/compile.c (hunk 1 above)
Comment (serhiy-storchaka) [reply Python/compile.c:3917]: It is not obvious, but `maybe_optimize_method_call()` can return 0, which means that an exception was set. `compiler_call()` should return 0 in this case.

**Comment by serhiy-storchaka** [comment]
Ping.

**Comment by brettcannon** [comment]
@ZackerySpytz just in case you didn't see the "ping" from Serhiy.

**Comment by ZackerySpytz** [comment]
I saw it.

**Comment by ZackerySpytz** [comment]
Hi, @serhiy-storchaka. Thank you for updating this PR. However, it seems that there is an error in the comment that you added. I will edit the comment.

**Comment by ZackerySpytz** [comment]
Hi, @serhiy-storchaka. I've created a backport PR for 3.7.
